internal/service/create/create.go

```diff
@@ -1,20 +1,206 @@
+	}
+
+	if componentArchiveService == nil {
+		return nil, fmt.Errorf("%w: componentArchiveService must not be nil", commonerrors.ErrInvalidArg)
+	}
+
+	if registryService == nil {
+		return nil, fmt.Errorf("%w: registryService must not be nil", commonerrors.ErrInvalidArg)
+	}
+
+	if moduleTemplateService == nil {
+		return nil, fmt.Errorf("%w: moduleTemplateService must not be nil", commonerrors.ErrInvalidArg)
+	}
+
+	if crdParserService == nil {
+		return nil, fmt.Errorf("%w: crdParserService must not be nil", commonerrors.ErrInvalidArg)
+	}
+
 	return &Service{
-		moduleConfigService: moduleConfigService,
+		moduleConfigService:     moduleConfigService,
+		gitSourcesService:       gitSourcesService,
+		securityConfigService:   securityConfigService,
+		componentArchiveService: componentArchiveService,
+		registryService:         registryService,
+		moduleTemplateService:   moduleTemplateService,
+		crdParserService:        crdParserService,
 	}, nil
 }
 
 func (s *Service) CreateModule(opts Options) error {
 	if err := opts.Validate(); err != nil {
 		return err
 	}
+	defer s.moduleConfigService.CleanupTempFiles()
+	moduleConfig, err := s.moduleConfigService.ParseAndValidateModuleConfig(opts.ModuleConfigFile)
+	if err != nil {
+		return fmt.Errorf("failed to parse module config: %w", err)
+	}
+
+	descriptor, err := componentdescriptor.InitializeComponentDescriptor(moduleConfig.Name, moduleConfig.Version)
+	if err != nil {
+		return fmt.Errorf("failed to populate component descriptor metadata: %w", err)
+	}
+
+	moduleResources, err := componentdescriptor.GenerateModuleResources(moduleConfig.Version, moduleConfig.ManifestPath,
+		moduleConfig.DefaultCRPath, opts.RegistryCredSelector)
+	if err != nil {
+		return fmt.Errorf("failed to generate module resources: %w", err)
+	}
+
+	if opts.GitRemote != "" {
+		if err = s.gitSourcesService.AddGitSources(descriptor, opts.GitRemote,
+			moduleConfig.Version); err != nil {
+			return fmt.Errorf("failed to add git sources: %w", err)
+		}
+	}
+	if moduleConfig.Security != "" && opts.GitRemote != "" {
+		err = s.configureSecScannerConf(opts, moduleConfig, descriptor)
+		if err != nil {
+			return fmt.Errorf("failed to configure security scanners: %w", err)
+		}
+	}
+
+	opts.Out.Write("- Creating component archive\n")
+	archive, err := s.componentArchiveService.CreateComponentArchive(descriptor)
+	if err != nil {
+		return fmt.Errorf("failed to create component archive: %w", err)
+	}
+	if err = s.componentArchiveService.AddModuleResourcesToArchive(archive,
+		moduleResources); err != nil {
+		return fmt.Errorf("failed to add module resources to component archive: %w", err)
+	}
+
+	if opts.RegistryURL != "" {
+		return s.pushImgAndCreateTemplate(opts, archive, moduleConfig)
+	}
+	return nil
+}
+
+func (s *Service) pushImgAndCreateTemplate(opts Options, archive *comparch.ComponentArchive, moduleConfig *contentprovider.ModuleConfig) error {
```

Comment:
Applies to the other funcs below as well... To me it looks a bit weird that opts is the first argument. Usually, it is the last one... Any reason for having it at the beginning?